cmd/snapd/main.go

```diff
@@ -42,25 +45,54 @@ func init() {
 	errtracker.SnapdVersion = cmd.Version
 }
 
+// commandline args
+var opts struct {
+	User bool `long:"user" description:"Start the user session instance of snapd"`
+}
+
+// Daemon describes all operations necessary to deal with a daemon instance
+type Daemon interface {
+	Init() error
+	Start()
+	Stop() error
+	Dying() <-chan struct{}
+	SetVersion(version string)
+}
+
 func main() {
 	cmd.ExecInCoreSnap()
-	if err := run(); err != nil {
+
+	parser := flags.NewParser(&opts, flags.HelpFlag|flags.PassDoubleDash|flags.PassAfterNonOption)
+	_, err := parser.ParseArgs(os.Args[1:])
+	if err != nil {
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		os.Exit(1)
 	}
-}
 
-func run() error {
-	httputil.SetUserAgentFromVersion(cmd.Version)
+	var d Daemon
+	if opts.User {
+		d, err = user.NewDaemon()
+	} else {
+		httputil.SetUserAgentFromVersion(cmd.Version)
```

Comment:
Can we call `SetUserAgentFromVersion` unconditionally? I suspect we can and it would not hurt in any way.